services/uploads/src/clamav.js

```diff
@@ -1,25 +1,32 @@
 /* eslint-disable no-console */
 
-const AWS = require("aws-sdk");
+const {
+  S3Client,
+  DeleteObjectsCommand,
+  GetObjectCommand,
+  ListObjectsV2Command,
+  PutObjectCommand,
+} = require("@aws-sdk/client-s3");
 const fs = require("fs");
 const child_process = require("child_process");
 const path = require("path");
 const constants = require("./constants");
 const utils = require("./utils");
+const { pipeline } = require("stream/promises");
 
-const S3 = new AWS.S3();
+const S3 = new S3Client();
 
 /**
  * Lists all the files from a bucket
  *
  * returns a list of keys
  */
 async function listBucketFiles(bucketName) {
+  const listObjects = new ListObjectsV2Command({
+    Bucket: bucketName,
+  });
   try {
-    const listFilesResult = await S3.listObjectsV2({
-      Bucket: bucketName,
-    }).promise();
-
+    const listFilesResult = await S3.send(listObjects);
```

Comment:
S3's List Objects is one of those sneaky methods that may truncate the result list and give you a continuation token. If there's a chance there will ever be more than 1000 files in this bucket, we'll need to check for the presence of `listFilesResult.NextContinuationToken`.

Reply:
@benmartin-coforma Good concern! I had not initially considered this, however it looks like there's no need to worry:
This function is only used to list the clamav definition files, which at the moment is five (only four of which get updated now, I think `daily.cld` is no longer used). These files get overwritten when there is an update, so they should never be more than this, plus or minus one
![Screenshot 2023-11-15 at 9 24 10 AM](https://github.com/Enterprise-CMCS/macpro-mdct-mcr/assets/57802560/9f8bb265-e723-46ca-8e68-e7a1fb90da35)

Additionally I checked with the CLI to see if it lists versions, which could make it lengthy. It does not. It only lists the latest version, so again, five items
`aws s3api list-objects-v2 --bucket uploads-main-avscan-446712541566`
![Screenshot 2023-11-15 at 9 26 31 AM](https://github.com/Enterprise-CMCS/macpro-mdct-mcr/assets/57802560/1d8763f1-7c62-4443-ba98-68e5d6d08e41)

Reply:
Okay, sounds good!